apis/apps/v1alpha1/sidecarset_types.go

```diff
@@ -43,6 +43,10 @@ type SidecarSetSpec struct {
 // SidecarContainer defines the container of Sidecar
 type SidecarContainer struct {
 	corev1.Container `json:",inline"`
+	// the sidecar container resource requirement that is proportional to the pod total resource
+	// this will override the corresponding resources fields in the container with the same resourceName
+	// the key of the map is the percentage
+	ProportionalResource map[corev1.ResourceName]uint8 `json:"proportionalResource,omitempty"`
```

Comment:
- This field should be optional.
- use int32 instead of uint8

You should explain clearly what the int value should be. E.g., it is a value from 1 - 100. 

One more thought, maybe the algorithm of calculating the size should be `size = max(container.spec.request,  main-containers.request * x /100)`

Reply:
> * This field should be optional.

isn't this optional already?

Reply:
> * use int32 instead of uint8

try to save some space, is there any advantage of using int32?

Reply:
> e size should be `size = max(container.spec.request, main-containers.request * x /100)`

What is "main" here?  I don't fully get the idea and it could be too complicated for general users.